lnwallet/chancloser/chancloser.go

```diff
@@ -837,6 +873,175 @@ func (c *ChanCloser) proposeCloseSigned(fee btcutil.Amount) (*lnwire.ClosingSign
 	return closeSignedMsg, nil
 }
 
+// handleRemoteProposal sanity checks the remote's ClosingSigned message and
+// tries to determine an acceptable fee to reply with. It may return:
+// - a message and no error (continue negotiation)
+// - no message and an error (negotiation failed)
+// - no message and no error (indicating we agree with the remote's fee)
+func (c *ChanCloser) handleRemoteProposal(
+	remoteMsg *lnwire.ClosingSigned) (lnwire.Message, error) {
+
+	isFunder := c.cfg.Channel.IsInitiator()
+
+	// If FeeRange is set, perform FeeRange-specific checks.
+	if remoteMsg.FeeRange != nil {
+		// If legacyNegotiation is already set, fail outright.
+		if c.legacyNegotiation {
+			return nil, ErrCloseTypeChanged
+		}
+
+		// Set rangeNegotiation to true.
+		c.rangeNegotiation = true
+
+		// Get the intersection of our two FeeRanges if one exists.
+		overlap := c.idealFeeRange.GetOverlap(remoteMsg.FeeRange)
+
+		if isFunder {
+			// If the fundee replies with a FeeRange, there must be
+			// overlap with our FeeRange.
+			//
+			// BOLT#02:
+			// - otherwise (it is not the funder)
+			//   - ...
+			//   - otherwise
+			//     - MUST propose a fee_satoshis in the overlap
+			//	 between received and (about-to-be) sent
+			//       fee_range.
+			if overlap == nil {
+				return nil, ErrNoRangeOverlap
+			}
+
+			// This is included in the above requirement.
+			if !overlap.InRange(remoteMsg.FeeSatoshis) {
+				return nil, ErrFeeNotInOverlap
+			}
+
+			// If the above checks pass, the funder must reply with
+			// the same fee_satoshis.
+			//
+			// BOLT#02:
+			// - if it is the funder:
+			//   - ...
+			//   - otherwise:
+			//     - MUST reply with the same fee_satoshis.
+			_, err := c.proposeCloseSigned(remoteMsg.FeeSatoshis)
+			if err != nil {
+				return nil, err
+			}
+
+			// Return nil values to indicate we are done with
+			// negotiation.
+			return nil, nil
+		}
+
+		// If we are the fundee and we have already sent a ClosingSigned
+		// to the funder, we should not be calling this function.
+		//
+		// BOLT#02:
+		// - if it is the funder:
+		//   - ...
+		//   - otherwise:
+		//     - MUST reply with the same fee_satoshis.
+		//
+		// Since the funder must reply with the same fee_satoshis, the
+		// calling function should not call into this negotiation
+		// function.
+		if c.lastFeeProposal != 0 {
+			return nil, ErrFeeRangeViolation
+		}
+
+		// If we are the fundee and there is no overlap between their
+		// fee_range and our yet-to-be-sent fee_range, send a warning.
+		//
+		// BOLT#02:
+		// - if there is no overlap between that and its own fee_range
+		//   - SHOULD send a warning.
+		//
+		// NOTE: The above SHOULD will probably be changed to MUST.
+		if overlap == nil {
+			warning := lnwire.NewWarning()
+			warning.ChanID = remoteMsg.ChannelID
+			warning.Data = lnwire.WarningData("ClosingSigned: no " +
+				"fee_range overlap")
+			return warning, nil
```

Comment:
Similar here re the responder just accepting w/e the funder wants in the case of no overlap.

Reply:
That's possible